packages/@uppy/aws-s3-multipart/src/index.js

```diff
@@ -19,9 +19,11 @@ function assertServerError (res) {
 }
 
 function throwIfAborted (signal) {
-  if (signal.aborted) { throw createAbortError('The operation was aborted', { cause: signal.reason }) }
+  if (signal?.aborted) { throw createAbortError('The operation was aborted', { cause: signal.reason }) }
 }
 
+const fileOnStart = new WeakMap()
```

Comment:
I find this name a bit confusing. We only want to store files on start? What happens afterwards?

Reply:
It's not to store the files on start, it's to start the `onStart` function for each file. Better naming suggestions welcome

Reply:
I'm still a bit confused about this pattern. Why are we saving function that gets and sets state? Can't we store the data we need and have a reusable function to do side-effects? How about saving the `s3Multipart` object per `file.id` and calling it `uploadStateCollection`?

Reply:
I'm also a bit confused why we need this function cache. Can't we just pass a callback function into the HTTPCommunicationQueue, something like this:

```
constructor() {
  const onMultipartStateChange = (...args) => this.#setS3MultipartState(..args)

  this.#companionCommunicationQueue = new HTTPCommunicationQueue(this.requests, this.opts, onMultipartStateChange)
}
```

then inside of the http queue we call that passed function instead of looking it up from the `stateUpdaterMap `?

Reply:
I think we need some sort of cache for when:
- you start a upload
- create multipart endpoint is called
- response is needed for continuation
- you then pause
- to resume you need to upload ID again, either from cache or calling the endpoint again

However, since this wasn't needed before, I wonder why it's needed now. AFAIK we already have upload state in uppy core with everything we need. It should be 'cached' already there?